### PR TITLE
Added an option to add callsign database URL for lookups on a map.

### PR DESCRIPTION
**Pull request by luarvique**

### CODE REVIEW

File: htdocs/map.js [hunk 1]
```diff
@@ -340,6 +344,15 @@ $(function(){
...
     }
 
+    var linkifyCallsign = function(callsign) {
+        if ((callsign_url == null) || (callsign_url == ''))
+            return callsign;
+        else
+            return '<a href="#" onclick="window.open(' + "'" +
+                callsign_url.replaceAll('{}', callsign) +
+                "','callsign_info'" + ');">' + callsign + '</a>';
+    };
```
Comment (jketterl) [inline htdocs/map.js:354]: Shouldn't this be possible without javascript? A simple `<a href="..." target="_blank">` normally does the trick.

File: owrx/config/defaults.py [hunk 2]
```diff
@@ -147,6 +147,7 @@
     squelch_auto_margin=10,
     google_maps_api_key="",
     map_position_retention_time=2 * 60 * 60,
+    callsign_url="https://www.qrzcq.com/call/{}",
```
Comment (jketterl) [inline owrx/config/defaults.py:150]: Personally, I'm not a big fan of any of the ham radio profile pages, so I'd prefer the default to be empty. I don't remember any previous requests for this particular feature, so I have no idea what the general opinion on this would be...

**Review by jketterl — CHANGES_REQUESTED** [review]

File: htdocs/map.js (hunk 1 above)
Comment (luarvique) [reply htdocs/map.js:354]: The change you are suggesting will open a new window every time. My original code will reuse the same window, once open.

File: owrx/config/defaults.py (hunk 2 above)
Comment (luarvique) [reply owrx/config/defaults.py:150]: The QRZCQ has been selected since it always shows at least the name and location of the operator. Another option would be QRZ, but it usually contains less public data. You are welcome to remove the default value completely or even insert "I AM NOT A BIG FAN" instead.

File: htdocs/map.js (hunk 1 above)
Comment (jketterl) [reply htdocs/map.js:354]: If you want to reuse the same tab, change `target` to a frame name (that is not one of the reserved targets).

File: owrx/config/defaults.py (hunk 2 above)
Comment (jketterl) [reply owrx/config/defaults.py:150]: You don't seem to be a big fan of discussions. Noted.

File: htdocs/map.js (hunk 1 above)
Comment (luarvique) [reply htdocs/map.js:354]: This apparently does not work on some browsers, but I have pushed the change you requested.

File: htdocs/map.js [hunk 3]
```diff
@@ -37,6 +37,7 @@ $(function(){
     var retention_time = 2 * 60 * 60 * 1000;
     var strokeOpacity = 0.8;
     var fillOpacity = 0.35;
+    var callsign_url = null;
```
Comment (jketterl) [inline htdocs/map.js:40]: I'm assuming that the "unconfigured" value would be an empty string. If so, it's probably best to set this to an empty string, too.

File: htdocs/map.js (hunk 1 above)
Comment (jketterl) [reply htdocs/map.js:354]: It would surprise me, but if that's the case then it probably will not matter in this particular instance: The link that leads to the map is done the same way.

File: htdocs/map.js (hunk 3 above)
Comment (luarvique) [reply htdocs/map.js:40]: As I previously mentioned, set it to anything you like.

Comment (jketterl) [reply htdocs/map.js:40]: Not necessarily talking about the default value here. Just trying to simplify the code: If this was an empty string from the beginning, you would not need to test for `null` later on (line 348 RN)

Comment (luarvique) [reply htdocs/map.js:40]: Oh, that. I always test optional pointers for null. If you are only testing for empty string, you are setting yourself up for a crash in the future.

Comment (jketterl) [reply htdocs/map.js:40]: This is javascript, I don't think the concept of pointers exists here, at least not in the code.

Either way, I don't mind the extra check, it certainly won't hurt.
